### ISSUE TITLE
warning - not compatible with Anaconda python 3.6

### ISSUE THREAD

**Issue by tingyuansen** [jobovy/galpy#321]
Hi Jo,

It seems like for Anaconda python 3.6, the "warning" input is incompatible with galpy. I bet it is a simple fix. For now, I just removed all warning from the codes.

warnings.warn("interppotential_c extension module not loaded, because galpy_interppotential_c%s image was not found" % _ext_suffix,
---> 39                       galpyWarning)
TypeError: _warning() takes from 1 to 4 positional arguments but 6 were given

Cheers,
Yuan-Sen

**Comment by jobovy** [comment]
Hi Yuan-Sen,

Is this using the lastest master version? (``python -c "import galpy; print(galpy.__version__)"`` should say ``1.3.dev``). I believe this was fixed in #309, but let me know if there are still problems with this (I'm mostly still on 3.5...)

**Comment by tingyuansen** [comment]
Hi Jo,

It is 1.3 dev.. Emm, 3.6 is just tricky. I will try to revert back to 3.5
for now.


Cheers,
Yuan-Sen

**Comment by jobovy** [comment]
Thanks for checking. I don't seem to be able to reproduce this problem. I did
```
conda create --name galpywarnings python=3.6 numpy scipy matplotlib ipython
source activate galpywarnings
```
then installed the latest version of galpy and ran
```
python -c "from galpy.potential import interpRZPotential"
```
which gives the warning without any issue. Do you know what might be different in your setup?

**Comment by tingyuansen** [comment]
Hi Jo,

I am not sure. I got a new macbook recently (hooray! new job!). I
downloaded anaconda from
https://www.anaconda.com/download/#macos

And then pip install galpy. And it gives the errors when reading the
warning.

But if I do conda create, it does not complain. Would you want to try the
anaconda installer and simply pip install? This seems like a method that
many people use and might be useful to get it working through this route.


Cheers,
Yuan-Sen

**Comment by jobovy** [comment]
Are you just doing 
```
pip install galpy
```
? I think that will install the latest release, v1.2, which is more than a year old at this point and wouldn't have this fix in it (but you say that you get 1.3.dev from the version output, so that would imply that you are not just pip installing the latest version).

So you are saying that using Anaconda from the MacOS installer doesn't work, but using the ``conda create`` command that I gave does work? That's very strange... I'm not sure I want to risk messing up my Anaconda install by downloading the MacOS installer myself...

Maybe you could take a look at the actual code that the faulty installation is using? If you do
```
python -c "from galpy import util; print(util.__file__)"
```
you should get the file that the code is actually using (although I'm not sure that works for pip...). *If* this is a readable file, can you check whether the _warning function is the latest one [here](https://github.com/jobovy/galpy/blob/master/galpy/util/__init__.py#L11). If it's not a readable file, then nevermind...

**Comment by tingyuansen** [comment]
Hi Jo,

Totally my bad. When we iterated just now, I reinstalled python 2.7 to get
a few actions calculated. I thought your codes will revert me back to 3.6,
but apparently, it did not. The warning part works fine for python 2.7
(which was how I built my codes around).

But if I reinstall anaconda 3.6, and run the conda -create test, it gives
the following errors. It does seem like in python 3.6, the warnings
package does not work as before.

Belkin52:~ yting$ python -c "from galpy.potential import interpRZPotential"

Traceback (most recent call last):

  File "<string>", line 1, in <module>

  File
"/Users/yting/anaconda/lib/python3.6/site-packages/galpy/potential.py",
line 3, in <module>

    warnings.warn("A major change in versions > 1.1 is that all
galpy.potential functions and methods take the potential as the first
argument; previously methods such as evaluatePotentials, evaluateDensities,
etc. would be called with (R,z,Pot), now they are called as (Pot,R,z) for
greater consistency across the codebase",galpyWarning)

  File "/Users/yting/anaconda/lib/python3.6/warnings.py", line 99, in
_showwarnmsg

    msg.file, msg.line)

TypeError: _warning() takes from 1 to 4 positional arguments but 6 were
given

**Comment by jobovy** [comment]
Did you check the file from 
```
python -c "from galpy import util; print(util.__file__)"
```
? This error just doesn't make much sense, because it was fixed in 5d298e9147afb6934c08e228c3aa4e965367a235 and I can't replicate it in python 3.6.

**Comment by tingyuansen** [comment]
Hi Jo,

I am also out of idea here. For

python -c "from galpy import util; print(util.__file__)"


It shows that I am using python 3.6

python -c "from galpy import util; print(util.__file__)"

/Users/yting/anaconda/lib/python3.6/site-packages/galpy/util/__init__.py

But does your edit actually cause the problem? i.e., instead of having
1-4 positional inputs, now inputs 6? I think a clean install of
Anaconda does not interfere with your own Anaconda. I have two
Anaconda co-exist nicely... It could just be me missing something
obvious here.


Cheers,

Yuan-Sen

**Comment by jobovy** [comment]
Okay, I installed the MacOSX Anaconda for Python 3.6 and I don't get this error. I installed it, activated the correct environment and did
```
pip install -U --no-deps git+git://github.com/jobovy/galpy.git#egg=galpy
```

then I can run galpy without any problem.

Have you looked at the ``Users/yting/anaconda/lib/python3.6/site-packages/galpy/util/__init__.py`` file? Does it have the ``line=`` and ``file=`` inputs to ``_warning``?

**Comment by tingyuansen** [comment]
Hi Jo,

I think I figured this out. I mixed two problems.

The warning issue is due to version 1.2, and you are right that 1.3 dev
does not have this problem.

Another issue is more intricate but it is due to the fact that I did not
have a fresh installation of Anaconda, instead, I migrated my old
macbook system to the new one through "migration assistant".

You are right that, with a fresh installation, anaconda + galpy 1.3 should
work fine. Sorry for the confusion!


Cheers,
